### DISCUSSION TITLE
regarding callback function of any request permission in android studio for ANE

### DISCUSSION THREAD

**Discussion by balaedit** [airsdk/Adobe-Runtime-Support#766] [General]
I am creating ANE for my project. I was successfully created an ANE of requesting any permission but I didn't know how to handle/write the onRequestPermissionsResult handle for my ANE.

My Android ANE context function file as below : 

public FREObject call(FREContext freContext, FREObject[] freObjects) {
    Context context = freContext.getActivity().getApplicationContext();
    CustomExtensionContext adec = (CustomExtensionContext) freContext;
    android.app.Activity activity = adec.getActivity();
    adec.activity = activity;
    int isPermissionGranted = 0;
    FREObject retVal = null;

    if (context.checkSelfPermission(Manifest.permission.READ_PHONE_STATE) != PackageManager.PERMISSION_GRANTED) {
        activity.requestPermissions(new String[]{Manifest.permission.READ_PHONE_STATE},0);
        isPermissionGranted = 0;  // user can 'Allow' or 'deny' here /// need callback function // how to handle or write.
    } else {
        isPermissionGranted = 1;
    }

    try {
        retVal = FREObject.newObject(isPermissionGranted);
    } catch (FREWrongThreadException e) {
        e.printStackTrace();
    }
    return retVal;
}

The above line is returning the isPermissionGranted false for the first time as I had not handled the onRequestPermissionsResult.

Please, Help me how to write the onRequestPermissionResult in this file.

**Comment by marchbold** [comment]
You need to do this in your own activity and override the `onRequestPermissionResult` there.

Or use our Permissions ANE: https://airnativeextensions.com/extension/com.distriqt.Permissions

**Comment by balaedit** [comment]
requested you to give idea how to write the onRequestPermissionsResult in my activity file.

Thanks.

**Reply by balaedit** [reply]
I had finally succeeded creating ANE for onRequestPermissionsResult function in my custom ANE.

I also created ANE for device unique id and toast ane.

I am enjoying creating ANE and implementing native functionality in my custom ANE.

Thanks Andrew Frost and Michael.
